src/zenml/stack/stack.py

```
@@ -530,6 +531,21 @@ def setting_classes(self) -> Dict[str, Type["BaseSettings"]]:
                 setting_classes[key] = component.settings_class
         return setting_classes
 
+    @property
+    def requires_remote_server(self) -> bool:
+        """If the stack requires a remote ZenServer to run.
+
+        This is the case if any code is getting executed remotely. This is the
+        case for both remote orchestrators as well as remote step operators.
+
+        Returns:
+            If the stack requires a remote ZenServer to run.
+        """
+        return self.orchestrator.config.is_remote or (
```

In what other cases do you think in the future we can throw this error?